app/src/main/java/com/projectit3048c/MainActivity.kt

```diff
@@ -167,12 +167,10 @@ class MainActivity : ComponentActivity() {
                 onClick = {
                     var specimen = FoodAmount().apply {
                         foodName = inFoodName
-                        foodId = selectedFood?.let(){
-                            it.id
-                        } ?: 0
+                        foodId = selectedFood?.id ?: 0
```

Comment:
Don't really need a let scoped function when your just trying to get selectedFoodID. Use let when you have a code block with multiple methods in it.
https://kotlinlang.org/docs/scope-functions.html#let

Reply:
Covered in a previously submitted code review.  When that occurs, only first in gets credit.